src/main/kotlin/uk/gov/dluhc/printapi/messaging/service/AbstractEmailSender.kt

```diff
@@ -0,0 +1,30 @@
+package uk.gov.dluhc.printapi.messaging.service
+
+import uk.gov.dluhc.printapi.client.ElectoralRegistrationOfficeManagementApiClient
+import uk.gov.dluhc.printapi.database.entity.Certificate
+import uk.gov.dluhc.printapi.database.entity.PrintRequest
+import uk.gov.dluhc.printapi.messaging.models.ProcessPrintResponseMessage
+
+abstract class AbstractEmailSender(
+    private val electoralRegistrationOfficeManagementApiClient: ElectoralRegistrationOfficeManagementApiClient,
+) {
+    /**
+     * @param printResponse the printResponse that could trigger an email being sent
+     * @param certificate the certificate associated to the printResponse
+     * @return true if an email was sent otherwise false
+     */
+    abstract fun testAndSend(
```

Comment:
I'm not sure about the name. I was initially alarmed by the thought of "test" code in production, but it's not that kind of test 😆 

Perhaps it would be clearer and quicker to grasp if you did the check in the client, before calling this method. For example:

```
if (printResponse.statusStep != ProcessPrintResponseMessage.StatusStep.NOT_MINUS_DELIVERED) {
   certificateNotDeliveredEmailSender.send(printResponse, certificate)
}
```